src/components/analytics/index.ts

```diff
@@ -23,5 +45,44 @@ if (typeof GOOGLE_ANALYTICS_ID === 'string') {
       : 'mobileRegular',
   })
 } else {
-  ReactGA.initialize('test', { gtagOptions: { debug_mode: true } })
+  googleAnalytics.initialize('test', { gtagOptions: { debug_mode: true } })
+}
+
+function reportWebVitals({ name, delta, id }: Metric) {
+  gaCommandSendTiming('Web Vitals', name, Math.round(name === 'CLS' ? delta * 1000 : delta), id)
 }
+
+// tracks web vitals and pageviews
+export function GoogleAnalyticsReporter({ location: { pathname, search } }: RouteComponentProps): null {
```

Comment:
This should not be exported. The Google-specific stuff stays in `/analytics`.

Reply:
If you need initialization, you could instead export a general `Reporter` that wraps a google analytics reporter, which seems to be what this is. It may be enough to just rename this to something generic.

Reply:
sgtm done